homeassistant/components/eafm/__init__.py

```diff
@@ -0,0 +1,43 @@
+"""UK Environment Agency Flood Monitoring Integration."""
+
+from homeassistant.helpers import device_registry as dr
+
+from .const import DOMAIN
+from .station import Station
+
+
+async def async_setup(hass, config):
+    """Set up devices."""
+    hass.data[DOMAIN] = {}
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up flood monitoring sensors for this config entry."""
+    station_key = entry.data["station"]
+
+    device_registry = await dr.async_get_registry(hass)
+    device_registry.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        identifiers={(DOMAIN, "station-id", station_key)},
+        name=entry.title,
+        manufacturer="https://environment.data.gov.uk/",
+        model="Real Time flood-monitoring API",
```

Comment:
@balloob are we ok with using device registry for api service providers?

Reply:
Yes. Let's go ahead and implement https://github.com/home-assistant/architecture/issues/360 here.

Reply:
13 thumbs up for that proposal, I should make it an ADR 😆